src/simulation/LoadGenerator.cpp

```diff
@@ -1485,7 +1536,9 @@ LoadGenerator::execute(TransactionFrameBasePtr txf, LoadGenMode mode,
     auto msg = txf->toStellarMessage();
     txm.mTxnBytes.Mark(xdr::xdr_argpack_size(*msg));
 
-    auto addResult = mApp.getHerder().recvTransaction(txf, true);
+    // Skip certain checks for pregenerated transactions
+    bool isLoadgenTx = (mode == LoadGenMode::PAY_PREGENERATED);
```

Comment:
Nit: I think the `isLoadgenTx` variable name is confusing here, since all TXs are coming from loadgen. Maybe rename to `isPregeneratedTx`?